aws_topology/stackstate_checks/aws_topology/resources/dynamodb.py

```diff
@@ -61,54 +90,97 @@ class DynamodbTableCollector(RegisteredResourceCollector):
-    CLOUDFORMATION_TYPE = 'AWS::DynamoDB::Table'
+    CLOUDFORMATION_TYPE = "AWS::DynamoDB::Table"
+
+    @set_required_access_v2("dynamodb:ListTagsOfResource")
+    def collect_tags(self, table_arn):
+        return self.client.list_tags_of_resource(ResourceArn=table_arn).get("Tags", [])
+
+    @set_required_access_v2("dynamodb:DescribeTable")
+    def collect_table_description(self, table_name):
+        return self.client.describe_table(TableName=table_name)
+
+    def construct_table_description(self, table_name):
+        return {
+            "Table": {
+                "TableName": table_name,
+                "TableArn": self.agent.create_arn("AWS::DynamoDB::Table", self.location_info, resource_id=table_name),
+            }
+        }
+
+    def collect_table(self, table_name):
+        # If data collection fails, construct the ARN
+        data = self.collect_table_description(table_name) or self.construct_table_description(table_name)
+        # If ARN creation fails, skip tag collection
+        if data.get("Table", {}).get("TableArn"):
+            tags = self.collect_tags(data["Table"]["TableArn"])
+        else:
+            tags = []
+        return TableData(table=data, tags=tags)
+
+    @set_required_access_v2("dynamodb:ListTables")
+    def collect_tables(self):
+        for table in [
+            self.collect_table(table_name)
+            for table_name in client_array_operation(self.client, "list_tables", "TableNames")
+        ]:
+            yield table
 
     def process_all(self, filter=None):
-        for page in self.client.get_paginator('list_tables').paginate():
-            for table_name in page.get('TableNames') or []:
-                self.process_table(table_name)
-
-    def process_table(self, table_name):
-        table_description_raw = self.client.describe_table(TableName=table_name)
-        table_description = make_valid_data(table_description_raw)
-        table_data = table_description['Table']
-        table_arn = table_data['TableArn']
-        table_tags = self.client.list_tags_of_resource(ResourceArn=table_arn).get('Tags') or []
-        table_data['Tags'] = table_tags
-        table_data['Name'] = table_arn
-        table_data.update(with_dimensions([{'key': 'TableName', 'value': table_name}]))
-        self.emit_component(table_arn, self.COMPONENT_TYPE, table_data)
-        latest_stream_arn = table_data.get('LatestStreamArn')
+        if not filter or "dynamodb" in filter:
+            for table_data in self.collect_tables():
+                try:
+                    self.process_table(table_data)
+                except Exception:
+                    pass
+
+    def process_one_table(self, table_name):
+        self.process_table(self.collect_table(table_name))
+
+    @transformation()
+    def process_table(self, data):
+        table = Table(data.table, strict=False)
+        table.validate()
+        description = table.Table
+        output = make_valid_data(description.to_primitive())
```

Comment:
Not sure if it's relevant here, but if you do `.to_primitive()` you lose fields that wasn't in the schema. Are there any fields that you're not explicitly setting in the schema that you want to send to StackState?

Reply:
If so `data` or `data.table` is probably better

Reply:
Correct, I used to_primitive() wrongly, fixed. Thanks!